src/typeInference/annotateAST.ts

```diff
@@ -34,43 +34,43 @@ export function annotate(ast: IAST): SequenceType | undefined {
 			const left = annotate(ast[1][1] as IAST);
 			const right = annotate(ast[2][1] as IAST);
 
-			return annotateAddOp(ast, left, right);
+			return annotateBinOp(ast, left, right, 'add');
 		case 'integerConstantExpr':
 			const integerSequenceType = {
 				type: ValueType.XSINTEGER,
 				mult: SequenceMultiplicity.EXACTLY_ONE,
 			};
 
-			insertAttribute(ast, integerSequenceType);
+			insertAttribute(ast, 'type', integerSequenceType);
 			return integerSequenceType;
 		case 'doubleConstantExpr':
 			const doubleSequenceType = {
 				type: ValueType.XSDOUBLE,
 				mult: SequenceMultiplicity.EXACTLY_ONE,
 			};
 
-			insertAttribute(ast, doubleSequenceType);
+			insertAttribute(ast, 'type', doubleSequenceType);
 			return doubleSequenceType;
 		case 'decimalConstantExpr':
 			const decimalSequenceType = {
 				type: ValueType.XSDECIMAL,
 				mult: SequenceMultiplicity.EXACTLY_ONE,
 			};
 
-			insertAttribute(ast, decimalSequenceType);
+			insertAttribute(ast, 'type', decimalSequenceType);
 			return decimalSequenceType;
 		case 'stringConstantExpr':
 			const stringSequenceType = {
 				type: ValueType.XSSTRING,
 				mult: SequenceMultiplicity.EXACTLY_ONE,
 			};
 
-			insertAttribute(ast, stringSequenceType);
+			insertAttribute(ast, 'type', stringSequenceType);
 			return stringSequenceType;
 		default:
 			for (let i = 1; i < ast.length; i++) {
 				annotate(ast[i] as IAST);
 			}
-			return { type: ValueType.XSERROR, mult: SequenceMultiplicity.EXACTLY_ONE };
```

Comment:
Why is the error removed btw?

Reply:
It's not really an error, it something I added at the start of the prototype to mark that something could not get annotated. However this changed to just returning undefined later and this snippet of code was still left in there.